tool/tsh/options.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/teleport/lib/utils"
+
+	"github.com/gravitational/trace"
+)
+
+// AllOptions is a listing of all known OpenSSH options.
+var AllOptions = map[string]map[string]bool{
```

Comment:
I don't see a reason for this map, simple switch statement with only supported values will work as well. Also, please let's use real constants for constant values, not strings.

Reply:
The requirement in #2330 is that we need to differentiate between options that are supported but not implemented where a warning is logged and options that are not supported at all where `tsh` will hard fail.